feat: create features component

System625 opened the pull request:
## 🔧 Refactor: Extract Features Section into Reusable Component

### 📋 Summary
This PR refactors the Features section from the main page component into a dedicated, reusable `Features` component to improve code organization, maintainability, and reusability.

### 🎯 What Changed
- **Extracted Features Section**: Moved the entire features section logic from `src/app/page.tsx` into a new dedicated component `src/components/Features.tsx`
- **Component Separation**: The features array, JSX structure, and all related styling are now encapsulated in their own component
- **Import Update**: Updated `src/app/page.tsx` to import and use the new `Features` component

### 🏗️ Technical Details

#### Files Modified:
- **`src/app/page.tsx`**: 
  - Removed inline features section code (~80 lines)
  - Added import: `import Features from "@/components/Features"`
  - Replaced inline JSX with `<Features />` component

#### Files Added:
- **`src/components/Features.tsx`**: 
  - New standalone component (89 lines)
  - Contains TypeScript interface definition for Feature type
  - Encapsulates features data array with 4 core features:
    - Smart Matching (AI-powered algorithm)
    - Integrated Communication (video chat & messaging)
    - Verified Profiles (vetted users)
    - Deal Analytics (comprehensive insights)
  - Maintains all original styling and responsive design
  - Includes dark mode support and hover animations
  
 ### 📸 Screenshots
 
<img width="1366" height="768" alt="Screenshot (1699)" src="https://github.com/user-attachments/assets/6c1f0125-3c03-428f-b8f3-6f18d3f1913b" />
<img width="1366" height="768" alt="Screenshot (1700)" src="https://github.com/user-attachments/assets/faa2b24e-23bb-4679-b88d-fc169f28ebb8" />


### ✅ Benefits
1. **Improved Code Organization**: Features logic is now isolated and self-contained
2. **Enhanced Reusability**: Features component can be easily reused across different pages
3. **Better Maintainability**: Changes to features section only require editing one dedicated file
4. **Reduced Page Component Size**: Main page component is now more focused and readable
5. **Component-Based Architecture**: Follows React best practices for component separation

### 🔍 Code Quality
- **Type Safety**: Added TypeScript interface for Feature objects
- **Consistent Styling**: Maintains all existing Tailwind CSS classes
- **Client-Side Component**: Properly marked with "use client" directive
- **Clean Imports**: Organized imports for UI components and icons

Closes #27

SudiptaPaul-31 commented:
<img width="1496" height="620" alt="image" src="https://github.com/user-attachments/assets/a1ec3b4d-de0a-4dd7-ab96-e1170df17309" />
@System625  please fix this

System625 commented:
Hi @SudiptaPaul-31  , I've fixed the build error

The root cause was the await connectDB() call in the root layout was blocking static page generation during build time.
I removed the database connection from src/app/layout.tsx:22 and made the layout synchronous. Now the build now completes successfully with all pages generating as static content.

Key Takeaway: Database connections should only be added in API routes or server components that actually need them, not in the root layout which affects all pages during static generation.

SudiptaPaul-31 reviewed (commented):
Good Work LGTM
